src/pudl/extract/ferc1.py

```diff
@@ -897,7 +922,94 @@ def extract_dbf_generic(
             "min_year": min(ferc1_settings.dbf_years),
             "max_year": max(ferc1_settings.dbf_years),
         },
-    )
+    ).assign(sched_table_name=table_name)
+
+
+def extract_xbrl_concat(
+    table_names: list[str],
+    period: Literal["duration", "instant"],
+    ferc1_engine: sa.engine.Engine,
+    ferc1_settings: Ferc1Settings,
+) -> pd.DataFrame:
+    """Combine multiple raw xbrl instant or duration tables into one.
+
+    Args:
+        table_names: The list of raw table names provided in TABLE_NAME_MAP_FERC1
+            under xbrl. These are the tables you want to combine.
+        period: Either duration or instant, specific to xbrl data.
+        ferc1_engine: An SQL Alchemy connection engine for the FERC Form 1 database.
+        ferc1_settings: Object containing validated settings relevant to FERC Form 1.
+            Contains the tables and years to be loaded into PUDL.
+
+    There are some instances where multiple xbrl tables ought to be combined into
+    a single table to best mesh with data from the other source. This function
+    concatinates those tables into one. It is similar to the extract_dbf_concat
+    except that this function handles the instant and duration tables from xbrl.
+
+    It does not combine instant and duration tables, rather, it creates an instant table
+    that is a combination of several other instant tables, and a duration table that is
+    the combination of several other instant tables (those listed in table_names).
+    """
+    tables = []
+    for raw_table_name in table_names:
+        tables.append(
+            extract_xbrl_generic(
+                table_name=raw_table_name,
+                period=period,
+                ferc1_engine=ferc1_engine,
+                ferc1_settings=ferc1_settings,
+            )
+        )
+
+    combined_table = pd.concat(tables)
+    if combined_table.empty:
+        combined_table = pd.DataFrame()
+    if not len(combined_table) == sum([len(x) for x in tables]):
```

Comment:
My understanding is that `AssertionErrors` are more robust than `assert` statements because you can "turn off" assert statements? I'm not super familiar with it, but I've defaulted to `AssertionErrors` generally because of that. 

I think you're probably right though that I'm being extra paranoid here and don't need this length check. If you add arguments to `concat` it can alter the number of rows, but just using a plain old `concat()` it shouldn't drop anything. I'm gonna remove this section.